src/Widgets/calendar/Grid.vala

```diff
@@ -260,6 +264,36 @@ namespace DateTime.Widgets {
         private uint day_hash (GLib.DateTime date) {
             return date.get_year () * 10000 + date.get_month () * 100 + date.get_day_of_month ();
         }
+        
+                
+        public void add_event_dots (E.Source source, Gee.Collection<ECal.Component> events) {
+            foreach (var component in events) {
+            
+                unowned ICal.Component ical = component.get_icalcomponent ();
+                ICal.Time? start_time = ical.get_dtstart ();
+                time_t start_unix = start_time.as_timet ();
+                var t = new GLib.DateTime.from_unix_utc (start_unix);
+                var d_hash = day_hash (t);
+                if (data.has_key (d_hash)) {
+                    data[d_hash].add_dots (source, component.get_icalcomponent ());
```

Comment:
You already have `component.get_icalcomponent ()` cached as `ical` right? Probably don't need to get it again